src/cleanvision/dataset/fsspec_dataset.py

```diff
@@ -18,11 +18,12 @@ def __init__(
         self,
         data_folder: Optional[str] = None,
         filepaths: Optional[List[str]] = None,
-        storage_opts: Dict[str, str] = {},
+        storage_opts: Optional[Dict[str, str]] = {},
     ) -> None:
         super().__init__()
         self.storage_opts = storage_opts
-        ignore_missing = self.storage_opts.pop("ignore_missing_keys", False)
+        if self.storage_opts:
```

Comment:
I think this if is not required with `False` being a default for the `pop` function?